packages/graphql-codegen-resolvers-models/src/index.ts

```diff
@@ -0,0 +1,166 @@
+import { printSchemaWithDirectives } from "@graphql-tools/utils";
+import ts, { factory, TypeAliasDeclaration } from "typescript";
+import { ParsedMapper } from "@graphql-codegen/visitor-plugin-common";
+import {
+  Types,
+  PluginFunction,
+  addFederationReferencesToSchema,
+} from "@graphql-codegen/plugin-helpers";
+import { parse, visit, GraphQLSchema, printSchema } from "graphql";
+import { ResolversModelsVisitor } from "./visitor";
+import { ResolversModelsPluginConfig, MapperConfigValue } from "./config";
+
+export const plugin: PluginFunction<
+  ResolversModelsPluginConfig,
+  Types.ComplexPluginOutput
+> = (
+  schema: GraphQLSchema,
+  documents: Types.DocumentFile[],
+  config: ResolversModelsPluginConfig,
+) => {
+  const transformedSchema = config.federation
+    ? addFederationReferencesToSchema(schema)
+    : schema;
+
+  if (!config.modelIntersectionSuffix || !config.mappers) {
+    return { content: "" };
+  }
+
+  const visitor = new ResolversModelsVisitor(config, transformedSchema);
+
+  const printedSchema = config.federation
+    ? printSchemaWithDirectives(transformedSchema)
+    : printSchema(transformedSchema);
+  const astNode = parse(printedSchema);
+  visit(astNode, { leave: visitor } as any);
+
+  const content = Object.entries(visitor.getValidMappers()).reduce(
+    (acc, [typeName, model]) => {
+      acc = [
+        ...acc,
+        ...getModelAST(
```

Comment:
Copying array here is conceptually correct but tends to affect performance significantly in practical cases with big lists.

For our use case it doesn't really matter, just a nit